crates/roc_std/src/roc_list.rs

```diff
@@ -56,7 +56,7 @@ impl<T> RocList<T> {
     }
 
     pub fn iter(&self) -> impl Iterator<Item = &T> {
-        self.into_iter()
+        self.as_slice().iter()
```

Comment:
We can change this back after reintroducing `IntoIterator` for `&'a RocList` (see my other comment!)